generator/src/test/java/com/graphicsfuzz/generator/tool/GenerateTest.java

```diff
@@ -373,4 +374,28 @@ public void testNoInjectionSwitchIfDisabled() throws Exception {
 
   }
 
+  @Test
+  public void testBeRobustWhenNoDonors() throws Exception {
+    final ShaderJobFileOperations fileOps = new ShaderJobFileOperations();
+    final String program = "#version 100\n" +
+        "void main() { }";
+    final String uniforms = "{}";
+    final File json = temporaryFolder.newFile("shader.json");
+    final File frag = temporaryFolder.newFile("shader.frag");
+    FileUtils.writeStringToFile(frag, program, StandardCharsets.UTF_8);
+    FileUtils.writeStringToFile(json, uniforms, StandardCharsets.UTF_8);
```

Comment:
Nit: fileOps has this method.

Reply:
Although we could change `fileOps.writeStringToFile` to use `StandardCharsets.UTF_8` instead of `defaultCharset`, so we are consistent.